### ISSUE TITLE
Inplace renaming of columns in c++ record batch.

### ISSUE THREAD

**Issue by ava6969** [apache/arrow#15161]
### Describe the usage question you have. Please include as many useful details as  possible.

I would love to rename the columns of record batch without creation of the record batch. Is that possible?

I tried this without no success.

record_batch = record_batch->ReplaceSchemaMetadata(std::make_shared<arrow::KeyValueMetadata>(new_columns));

### Component(s)

C++

**Comment by westonpace** [comment]
There is `arrow::Table::RenameColumns` and I would be in favor of a similar `arrow::RecordBatch::RenameColumns`.  However, this would still be creating a new record batch.

In Arrow-C++ classes such as `Array`, `RecordBatch`, `ChunkedArray`, and `Table` are immutable.  Changing this constraint would be pretty significant and is unlikely.  However, creating a new `RecordBatch` should be cheap because we do not copy the underlying buffers.  The cost should simply be a few `shared_ptr` allocations (possibly per column).

**Comment by ava6969** [comment]
I don't see this function in the record batch but i guess i can perform this operation myself. in what instance, is there a copy of the buffer.

**Comment by westonpace** [comment]
> I don't see this function in the record batch

Correct, one does not exist.  I was suggesting that someone create one.  If you create one yourself I'd encourage you to submit a PR.

> in what instance, is there a copy of the buffer.

Normally you have to go out of your way to copy a buffer.  For example, create an array builder, add all the values from the old array, and then build an array from that.

However, many of the compute functions (take, filter, sort, cast, etc.) will do this (some casts are zero-copy but not all).

**Comment by thisisnic** [comment]
This issue appears to be stale, so I'm going to close it for now. Please feel free to reopen if it's still relevant or you'd like to continue the discussion.
